pkg/k8s/identitybackend/identity_test.go

```diff
@@ -82,3 +92,119 @@ func (s *K8sIdentityBackendSuite) TestSanitizeK8sLabels(c *C) {
 		c.Assert(validation.ValidateLabels(selected, path), checker.DeepEquals, test.validationErrors)
 	}
 }
+
+type FakeHandler struct{}
+
+func (f FakeHandler) OnListDone()                                       {}
+func (f FakeHandler) OnAdd(id idpool.ID, key allocator.AllocatorKey)    {}
+func (f FakeHandler) OnModify(id idpool.ID, key allocator.AllocatorKey) {}
+func (f FakeHandler) OnDelete(id idpool.ID, key allocator.AllocatorKey) {}
+
+func getLabelsKey(rawMap map[string]string) allocator.AllocatorKey {
+	return key.GlobalIdentity{labels.Map2Labels(rawMap, labels.LabelSourceK8s).LabelArray()}
+}
+func getLabelsMap(rawMap map[string]string) map[string]string {
+	return getLabelsKey(rawMap).GetAsMap()
+}
+func createCiliumIdentity(id int, labels map[string]string) v2.CiliumIdentity {
+	return v2.CiliumIdentity{
+		ObjectMeta: v1.ObjectMeta{
+			Name: strconv.Itoa(id),
+		},
+		SecurityLabels: getLabelsMap(labels),
+	}
+}
+
+func TestGetIdentity(t *testing.T) {
+	simpleMap := map[string]string{"key": "value"}
+	simpleMap2 := map[string]string{"ke2": "value2"}
+	simpleMap3 := map[string]string{"key3": "value3"}
+	duplicateMap1 := map[string]string{"key": "foo=value"}
+	duplicateMap2 := map[string]string{"key=foo": "value"}
+
+	testCases := []struct {
+		desc         string
+		identities   []v2.CiliumIdentity
+		requestedKey allocator.AllocatorKey
+		expectedId   string
+	}{
+		{
+			desc:         "Simple case",
+			identities:   []v2.CiliumIdentity{createCiliumIdentity(10, simpleMap)},
+			requestedKey: getLabelsKey(simpleMap),
+			expectedId:   "10",
+		},
+		{
+			desc: "Multiple identities",
+			identities: []v2.CiliumIdentity{
+				createCiliumIdentity(10, simpleMap),
+				createCiliumIdentity(11, simpleMap2),
+				createCiliumIdentity(12, simpleMap3),
+			},
+			requestedKey: getLabelsKey(simpleMap2),
+			expectedId:   "11",
+		},
+		{
+			desc: "Duplicated key",
```

Comment:
What about when two Identities have the same label set?
In case 2 pods are deployed on two different nodes at the same time.
```
ID 10 -> {“app”: “foo”}
ID 11 -> {“app”: “foo”}
```
And then a new pod with the same label set is being created.

Previously, get() would return a random Identity, because Store.List() returns an unordered list.
Is it going to consistently get the same ID now?

Reply:
Now it returns first identity

Reply:
Added test